plugins/zify/g_zify.mlg

```diff
@@ -0,0 +1,76 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+{
+
+open Ltac_plugin
+open Stdarg
+open Tacarg
+open Pp
+
+let iter_hyps (tac:Tacinterp.Value.t) =
+  Proofview.Goal.enter  begin fun gl ->
+    let hyps = List.rev (Tacmach.New.pf_hyps_types gl) in
+
+    Tacticals.New.tclTHENLIST
+          (List.map (fun (h,t) -> Tacticals.New.tclTRY (Tacinterp.Value.apply tac [Tacinterp.Value.of_constr (EConstr.mkVar h) ; Tacinterp.Value.of_constr t])) hyps)
+    end
+
+let iter_specs (tac: Tacinterp.Value.t) =
+    Zify.iter_specs (fun c -> Tacinterp.Value.apply tac [Tacinterp.Value.of_constr c])
```

Comment:
Why have this in the mlg instead of implementing iter_specs fully in zify.ml?

Reply:
If I do so, I get an error (some unbound module) when loading the plugin from a .v file.

Reply:
>If I do so, I get an error (some unbound module) when loading the plugin from a .v file.

Would you make a branch where that problem occurs so I can have a look?

Reply:
It is there:
https://github.com/fajb/coq/tree/zify_loading_failure

I did not recompile from scratch but get the error:
File "./plugins/zify/ZifyInst.v", line 17, characters 0-32:
Error: while loading
/home/fbesson/sources/coq-fajb/plugins/zify/zify_plugin.cmxs:
no implementation available for Tacinterp

Reply:
It's `Ltac_plugin.Tacinterp` not just `Tacinterp` (note that Ltac_plugin is opened in the mlg)

Reply:
done.